lib/CMakeLists.txt

```diff
@@ -44,124 +38,173 @@ add_library(arango_static_strings STATIC
   Basics/StaticStrings.cpp
 )
 
+set(FROZEN_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/3rdParty/iresearch/external/frozen/include")
+set(FUNCTION2_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/3rdParty/iresearch/external/function2")
+
+add_library(arango_lightweight STATIC
+        ApplicationFeatures/ApplicationFeature.cpp
+        ApplicationFeatures/ApplicationServer.cpp
+        ApplicationFeatures/ShellColorsFeature.cpp
+        Basics/ArangoGlobalContext.cpp
+        Basics/CleanupFunctions.cpp
+        Basics/ConditionLocker.cpp
+        Basics/ConditionVariable.cpp
+        Basics/ErrorCode.cpp
+        Basics/Exceptions.cpp Basics/Exceptions.tpp Basics/Exceptions.h
+        Basics/FileResult.cpp
+        Basics/FileUtils.cpp
+        Basics/Guarded.h
+        Basics/Identifier.cpp
+        Basics/Mutex.cpp
+        Basics/NumberOfCores.cpp
+        Basics/PageSize.cpp
+        Basics/PhysicalMemory.cpp
+        Basics/ReadWriteLock.cpp
+        Basics/RebootId.cpp Basics/RebootId.h
+        Basics/Result.cpp
+        Basics/ResultError.cpp Basics/ResultError.tpp Basics/ResultError.h
+        Basics/StringBuffer.cpp
+        Basics/StringHeap.cpp
+        Basics/StringUtils.cpp
+        Basics/Thread.cpp
+        Basics/UnshackledMutex.cpp
+        Basics/application-exit.cpp
+        Basics/conversions.cpp
+        Basics/debugging.cpp
+        Basics/error.cpp
+        Basics/files.cpp
+        Basics/fpconv.cpp
+        Basics/levenshtein.cpp
+        Basics/memory.cpp
+        Basics/process-utils.cpp
+        Basics/signals.cpp
+        Basics/system-functions.cpp
+        Basics/tri-strings.cpp
+        Containers/ImmerMemoryPolicy.h
```

Comment:
Immer used only in replication2.
I know it was but maybe in future you moved it out?